test/core/handshake/server_ssl.c

```diff
@@ -140,9 +140,11 @@ static bool server_ssl_test(const char *alpn_list[], unsigned int alpn_list_len,
   SSL_load_error_strings();
   OpenSSL_add_ssl_algorithms();
 
-  const SSL_METHOD *method = TLSv1_2_client_method();
+  const SSL_METHOD *method = TLS_client_method();
   SSL_CTX *ctx = SSL_CTX_new(method);
-  if (!ctx) {
+  if (!ctx ||
```

Comment:
How about having a small static function like this:
`static int ssl_ctx_force_tls1_2(SSL_CTX *ctx)`?